massa-models/src/storage.rs

```diff
@@ -0,0 +1,32 @@
+use crate::prehash::Map;
+use crate::{Block, BlockId};
+use parking_lot::RwLock;
+use std::sync::Arc;
+
+pub struct StoredBlock {
+    pub block: Block,
+    pub serialized: Vec<u8>,
+}
+
+#[derive(Clone, Default)]
+pub struct Storage {
+    blocks: Arc<RwLock<Map<BlockId, Arc<RwLock<StoredBlock>>>>>,
+}
+
+impl Storage {
+    pub fn store_block(&self, block_id: BlockId, block: Block, serialized: Vec<u8>) {
+        // TODO: first check, and allow for, an already stored header for the block.
+        let stored_block = StoredBlock { block, serialized };
+        let to_store = Arc::new(RwLock::new(stored_block));
+        let mut blocks = self.blocks.write();
+        blocks.insert(block_id, to_store);
+    }
+
+    pub fn retrieve_block(&self, block_id: &BlockId) -> Option<Arc<RwLock<StoredBlock>>> {
```

Comment:
I have some concerns about the return type of this. If we want to make this interface be easily changeable the fact that each time we use this function and we have to make a `.read()` after is not generic. I haven't think if there is a simple possibility to make the change here but maybe worth some discussions.

Reply:
Regarding our discussion in https://github.com/massalabs/massa/issues/2344, the change would be at the level of the top-level map, not the individual block. So that will not affect it. 

Did you have another potential problem in mind?

Reply:
Oh yeah. My bad It's ok then :)